plugins/out_kinesis_streams/kinesis_api.c

```diff
@@ -276,8 +279,14 @@ static int process_event(struct flb_kinesis *ctx, struct flush *buf,
                          ctx->stream_name);
             return 2;
         }
-        /* guess space needed to write time_key */
-        len = 6 + strlen(ctx->time_key) + 6 * strlen(ctx->time_key_format);
+
+
+        /* transfer from const char * to char * */
+        tmp_time_key_format = (char *)calloc(strlen(ctx->time_key_format), sizeof(char));
+        strcpy(tmp_time_key_format, ctx->time_key_format);
```

Comment:
This is used to transfer from const char * to char * because some function like memcpy needs the argument to be char * instead of const char *. I was also confused by the formatting of the comments at first glance.